### PR TITLE
Add SQLFlow user guide

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Fix #572.

### CODE REVIEW

File: doc/user_guide.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
+# SQLFlow User Guide
+
+SQLFlow is a bridge that connects a SQL engine (e.g. MySQL, Hive, or MaxCompute) and TensorFlow and other machine learning toolkits.  SQLFlow extends the SQL syntax to enable model training and inference.
+
+## Overview
+
+Say you have your data in table `iris.train`. The first four columns are the features and the last column is the label.
```
Comment (typhoonzero) [inline doc/user_guide.md]: Maybe add a link to what "iris" is and what the features stand for.

File: doc/user_guide.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+```SQL
+SELECT * FROM iris.train
+TRAIN DNNClassifer
+WITH hidden_units = [10, 10], n_classes = 3, EPOCHS = 10
+COLUMN sepal_length, sepal_width, petal_length, petal_width
+LABEL class
+INTO sqlflow_models.my_dnn_model;
+```
+
+SQLFlow will parse the statement and transpile it to an equivalent Python program for you.
```
Comment (typhoonzero) [inline doc/user_guide.md:56]: transpile => translate ?

File: doc/user_guide.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+```SQL
+LABEL label_expr
+```
+
+- *label_expr* indicates the field name and the preprocessing method on the field content. e.g. `class`.
+
+Note: some field name may look like SQLFlow keywords. For example, the table may contain a field named label. You can use double quotes around the name `LABEL "label"` to work around the parsing error.
+
+### Save clause
```
Comment (typhoonzero) [inline doc/user_guide.md]: Into clause?

File: doc/user_guide.md [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+```SQL
+INTO table_references
+```
+
+- *table_references* indicates the table to save the trained model. e.g. `sqlflow_model.my_dnn_model`.
+
+Note: SQLFlow team is actively working on supporting saving model to third-party storage services such as AWS S3, Google Storage and Alibaba OSS.
+
+## Feature columns
```
Comment (typhoonzero) [inline doc/user_guide.md:164]: Actually, feature columns descriptions also include descriptions of feeding and processing input data (delimiter, shape).

File: doc/user_guide.md (hunk 1 above)
Comment (tonyyang-svail) [reply doc/user_guide.md]: Good point.

File: doc/user_guide.md (hunk 2 above)
Comment (tonyyang-svail) [reply doc/user_guide.md:56]: I feel "transpile" is better since SQLFlow does verification of the statement.

File: doc/user_guide.md (hunk 3 above)
Comment (tonyyang-svail) [reply doc/user_guide.md]: Sure.

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**